cmake/module/TryAppendLinkerFlag.cmake

```diff
@@ -0,0 +1,56 @@
+# Copyright (c) 2023 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+include(CheckCXXSourceCompiles)
+
+function(try_append_linker_flag flags_var flag)
+  cmake_parse_arguments(PARSE_ARGV 2
+    TALF                              # prefix
+    ""                                # options
+    "SOURCE;RESULT_VAR"               # one_value_keywords
+    "IF_CHECK_PASSED;IF_CHECK_FAILED" # multi_value_keywords
+  )
+
+  string(MAKE_C_IDENTIFIER "${flag}" result)
+  string(TOUPPER "${result}" result)
+  set(result "LINKER_SUPPORTS_${result}")
+  unset(${result})
+  if(CMAKE_VERSION VERSION_LESS 3.14)
+    set(linker_flags_var CMAKE_REQUIRED_LIBRARIES)
+  else()
+    set(linker_flags_var CMAKE_REQUIRED_LINK_OPTIONS)
+  endif()
+
+  if(MSVC)
+    set(${linker_flags_var} ${flag} /WX)
+  elseif(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
```

Comment:
- These are specific to the linker, not the platform.
- The Darwin platform is in flux (moving to lld)
- ld64 may be used on other platforms (BSDs)

So... I'm requesting as usual: please don't hard-code platforms like these. Test the flags and apply them if they work.

Reply:
Looking at our current impl, I see now you've mostly just ported over the old behavior.

Much of this seems quite janky to me, but I'll stop complaining about current behaviors. I might upstream a few changes though, and let you pick them up in a rebase.

Reply:
> Looking at our current impl, I see now you've mostly just ported over the old behavior.

Right. As we all were agree about that for the staging branch, IIRC.

> Much of this seems quite janky to me, but I'll stop complaining about current behaviors. I might upstream a few changes though, and let you pick them up in a rebase.

Great!

Reply:
>    * ld64 may be used on other platforms (BSDs)

FWIW, I've successfully tested this PR on FreeBSD 13.2 and OpenBSD 7.1 using the default toolchains.